app/assets/javascripts/discourse/app/components/post/menu/buttons/like.gjs

```diff
@@ -0,0 +1,139 @@
+import Component from "@glimmer/component";
+import { fn, hash } from "@ember/helper";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import icon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class PostMenuLikeButton extends Component {
+  static shouldRender(post) {
+    return post.showLike;
+  }
+
+  @service currentUser;
+
+  #element;
+
+  get disabled() {
+    return this.currentUser && !this.args.post.canToggleLike;
+  }
+
+  get title() {
+    // If the user has already liked the post and doesn't have permission
+    // to undo that operation, then indicate via the title that they've liked it
+    // and disable the button. Otherwise, set the title even if the user
+    // is anonymous (meaning they don't currently have permission to like);
+    // this is important for accessibility.
+
+    if (this.args.post.liked && !this.args.post.canToggleLike) {
+      return "post.controls.has_liked";
+    }
+
+    return this.args.post.liked
+      ? "post.controls.undo_like"
+      : "post.controls.like";
+  }
+
+  @bind
+  setElement(element) {
+    this.#element = element;
+  }
+
+  @action
+  animateToggle() {
+    this.#element.classList.add("has-like");
+    this.#element.querySelector(`.d-icon`).classList.add("heart-animation");
+  }
```

Comment:
Storing a reference and then manipulating later is a bit of a code smell.

I think a more idiomatic way would be to do something like

```js
@tracked animating;

@action
animateToggle(){
  this.animating = true;
}
```

and then put the class logic in the hbs like

```hbs
class={{if this.animating "has-like heart-animation"}}
```

Reply:
Done